### PR TITLE
Map / rename some more Particle things

### PR DESCRIPTION
**Pull request by copygirl**
In addition to naming a bunch of fields / methods / parameters I also changed particle names to what made more sense to me. Making this a PR in case anyone sees errors or has any improvement suggestions or general thoughts.
- Some particles seem to define their own, identical `age/maxAge` fields so I've named them `age_/maxAge_`. No idea if there's any better name.
- `ParticleSomething` : `FactorySomething`, `FactoryHappy` - Unsure how to name these. `FactorySomething` is used both for `DEPTH_SUSPEND` and `TOWN_AURA`.
- `ParticleCrack` - This is used to render parts of an icon for eating, tools breaking, snowballs, slimes and similar.
- `boa` (appears to have something to do with damage particles) and `bnr` (base for `EndRod` and `FireworksSpark`) are still unnamed.

### CODE REVIEW

**Review by Darkhax — APPROVED** [review]

File: mappings/net/minecraft/client/particle/ParticleFootstep.mapping [hunk 1]
```diff
@@ -10,8 +10,19 @@ CLASS none/bnc net/minecraft/client/particle/ParticleFootstep
 			ARG 6 velocityY
 			ARG 7 velocityZ
 			ARG 8 extra
+	FIELD L maxAge_ I
+	FIELD M textureManager Lnone/bxf;
+	FIELD a TEX Lnone/kp;
+	FIELD b age_ I
```
Comment (exaskye) [inline mappings/net/minecraft/client/particle/ParticleFootstep.mapping:16]: Why the extra underscore?

File: mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping [hunk 2]
```diff
@@ -1,7 +1,7 @@
-CLASS none/bni net/minecraft/client/particle/ParticleLava
+CLASS none/bni net/minecraft/client/particle/ParticleLavaEmber
```
Comment (exaskye) [inline mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping:1]: Just `Ember`?

**Review by exaskye — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping (hunk 2 above)
Comment (copygirl) [inline mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping]: > Some particles seem to define their own, identical age/maxAge fields so I've named them age_/maxAge_. No idea if there's any better name.

File: mappings/net/minecraft/client/particle/ParticleFootstep.mapping (hunk 1 above)
Comment (copygirl) [reply mappings/net/minecraft/client/particle/ParticleFootstep.mapping:16]: > Some particles seem to define their own, identical age/maxAge fields so I've named them age_/maxAge_. No idea if there's any better name.

File: mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping (hunk 2 above)
Comment (copygirl) [reply mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping:1]: I've grouped water related things as `WaterSomething` so this seemed to make sense to me.
(This should be the particle that shoots out of lava in an arc and spawns smoke particles as it goes, aka Mojang's particle identifier `lava`.)

File: mappings/net/minecraft/client/particle/ParticleFootstep.mapping (hunk 1 above)
Comment (exaskye) [reply mappings/net/minecraft/client/particle/ParticleFootstep.mapping:16]: Oh, sorry. Saw the diff in my email and didn't read the comment.

File: mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping (hunk 2 above)
Comment (exaskye) [reply mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping:1]: Visually it's an ember though, and a number of mods use it as such.

Doesn't really matter, either way.

**Review by exaskye — APPROVED** [review]

File: mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping (hunk 2 above)
Comment (copygirl) [reply mappings/net/minecraft/client/particle/ParticleLavaEmber.mapping:1]: Wouldn't mind going with it, though I lean towards `LavaEmber`.

:+1: / :-1: on using `Ember`?

**Pull request merged**